examples/2d/compute_speed.py

```diff
@@ -25,13 +25,39 @@
 # Certain backends are also GPU-only, we we want to detect that before running
 # the benchmark.
 
-import kymatio.scattering2d.backend as backend
+backends = []
+
+try:
+    if torch.cuda.is_available():
+        from skcuda import cublas
+        import cupy
+        from kymatio.scattering2d.backend import torch_skcuda_backend
+        backends.append(torch_skcuda_backend)
+except:
+    pass
+
+try:
+    from kymatio.scattering2d.backend import torch_backend
+    backends.append(torch_backend)
+except:
+    pass
+
+
+###############################################################################
+# Determine which devices (CPU or GPU) that are supported by the current
+# backend.
+
+if torch.cuda.is_available():
+    devices = ['cuda', 'cpu']
+else:
+    devices = ['cpu']
+
 
 ###############################################################################
-# Finally, we import the `Scattering2D` class that computes the scattering
+# Finally, we import the `Scattering2DTorch` class that computes the scattering
```

Comment:
We need a default to be occupied by `Scattering2D`. 

I think we can give it dynamic inheritance and say `Scattering2D(frontend='numpy')` or whatever is available by default. (Config file would be optional and not necessary). This will make the examples remain usable and `Scattering2D` do the default thing you want to have happen in most cases.

Reply:
sounds good

Reply:
Note that this is possible in two ways - I am not sure which is better:
1.
```python
class Scattering2D(object):
    def __init__(self, frontend):
        if frontend == 'pytorch':
            from . import Scattering2DTorch
            self.__base__ = Scattering2DTorch
            super(Scattering2D, self).__init__()
    # other stuff
```

or

```python
class Scattering2D(object):
    def __init__(self, frontend):
        if frontend == 'pytorch':
            from . import Scattering2DTorch
            self.__class__ = Scattering2DTorch
            Scattering2DTorch.__init__(self) # maybe
    # other stuff
```

In the first case, you dynamically inherit, in the second case, you dynamically _become_ that class. What do people think? Semantically probably the second is better?

Reply:
fixed